### PR TITLE
Ali GOK React Week2

### PR DESCRIPTION
**Pull request by aligoknl**

### CODE REVIEW

**Comment by crevulus** [comment]
@aligoknl You've left out what is probably the most important file of any repository: the `package.json`! This file describes the dependencies, node configuration, linting rules, supported browsers, and all sorts of other useful info. Please commit it, remove that line from your `.gitignore`, and then I will be able to run your application!

**Comment by aligoknl** [comment]
> @aligoknl You've left out what is probably the most important file of any repository: the `package.json`! This file describes the dependencies, node configuration, linting rules, supported browsers, and all sorts of other useful info. Please commit it, remove that line from your `.gitignore`, and then I will be able to run your application!

I added it. Sorry for the trouble.

File: week2/project/ecommerce/src/components/Error.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const Error = () => {
+  return <p>Sorry there is an error. We could not find the products.</p>;
```
Comment (crevulus) [inline week2/project/ecommerce/src/components/Error.js:2]: You have used the same error component with the same message for both API calls. It should really say something else when the categories endpoint fails. Perhaps you could make this component more dynamic? 🟠

File: week2/project/ecommerce/src/FetchData.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+const FetchData = (url, setData) => {
```
Comment (crevulus) [inline week2/project/ecommerce/src/FetchData.js:3]: Wow, you are a week early using hooks! Impressive.
For hooks, we follow the naming conventions:

1. Start the name of the hook with use. `useFetchData`.
2. Name the file the same as the hook.
3. Don't use Title Case or kebab-case.
🟢

File: week2/project/ecommerce/src/App.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import CategoryList from "./components/CategoryList.js";
+import Products from "./components/Products.js";
+import FetchData from "./FetchData";
+
+const App = () => {
+  const [category, setCategory] = useState("allCategories");
+  const [categories, setCategories] = useState([]);
+  const URL = "https://fakestoreapi.com/products/categories";
+
+  const { loading, error } = FetchData(URL, setCategories);
```
Comment (crevulus) [inline week2/project/ecommerce/src/App.js:12]: Very nice!

File: week2/project/ecommerce/src/components/ProductDetails.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import FetchData from "../FetchData";
+import Loading from "./Loading";
+import Error from "./Error";
+
+const ProductDetails = () => {
+  const [product, setProduct] = useState({});
+  const { id } = useParams();
+  const URL_item = `https://fakestoreapi.com/products/${id}`;
```
Comment (crevulus) [inline week2/project/ecommerce/src/components/ProductDetails.js:10]: Variable naming convention is to use camelCase for variables inside the function/component or CAPITAL_SNAKE_CASE for variables outside the component. 🟢

File: week2/project/ecommerce/src/components/CategoryList.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { useState } from "react";
+import Loading from "./Loading";
+import Error from "./Error";
+
+function CategoryList({ allCategories, setCategory, loading, error }) {
+  const [activeButton, setActiveButton] = useState(null);
+
+  const toggle = (index, item) => {
+    if (activeButton === index) {
+      setCategory("allCategories");
```
Comment (crevulus) [inline week2/project/ecommerce/src/components/CategoryList.js:10]: Why do you need both a `category` and an `activeButton` variable? Is there a way you could check which button is active based on the `category` string? 🟠

File: week2/project/ecommerce/src/App.css [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+.category-item:hover {
+  background-color: #dcdceb;
+  cursor: pointer;
+}
+.category-item-selected {
+  background-color: #b1b1c1;
+  padding: 0.4rem;
+  margin: 2px;
+  cursor: pointer;
```
Comment (crevulus) [inline week2/project/ecommerce/src/App.css:28]: Does this do anything differently to the `cursor:pointer` on line 22? 🟢

File: week2/project/ecommerce/src/App.css [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  .item {
+    padding: 1rem;
+  }
+}
+
+@media (min-width: 1000px) {
+  .products {
+    display: grid;
+    grid-template-columns: repeat(3, 1fr);
```
Comment (crevulus) [inline week2/project/ecommerce/src/App.css:93]: Smart use of css grid!

**Review by crevulus — COMMENTED** [review]
Excellent work, Ali! I have very few comments for you and zero 🔴 comments. Fantastic! Keep up the good work.

File: week2/project/ecommerce/src/App.css (hunk 6 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/App.css:28]: It is not necessary, you're right. I have removed.

File: week2/project/ecommerce/src/App.css (hunk 7 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/App.css:93]: Thank you, I like grid!

File: week2/project/ecommerce/src/App.js (hunk 3 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/App.js:12]: Thanks!

File: week2/project/ecommerce/src/FetchData.js (hunk 2 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/FetchData.js:3]: I have changed it as useFetchData.

File: week2/project/ecommerce/src/components/ProductDetails.js (hunk 4 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/components/ProductDetails.js:10]: Indeed. I have changed as `url_item.`

File: week2/project/ecommerce/src/components/CategoryList.js (hunk 5 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/components/CategoryList.js:10]: I used it also for the CSS. I am changing the CSS class. I would like to keep it like that if is okay.

File: week2/project/ecommerce/src/components/Error.js (hunk 1 above)
Comment (aligoknl) [reply week2/project/ecommerce/src/components/Error.js:2]: Yes, thank you for the suggestion. I have made it more dynamic now.

**Comment by aligoknl** [comment]
> Excellent work, Ali! I have very few comments for you and zero 🔴 comments. Fantastic! Keep up the good work.

Thank you so much, Chris. I have updated my code according to your valuable comments.

File: week2/project/ecommerce/src/components/CategoryList.js (hunk 5 above)
Comment (crevulus) [reply week2/project/ecommerce/src/components/CategoryList.js:10]: In your `activeToggle` function, where you check the `activeButton` index against the map item index, you could also check if `category === item`, no?
You can leave this how it is, of course, but it's important for you to get into the habit of not using too many data points, or it quickly becomes difficult to manage.

**Comment by crevulus** [comment]
Great stuff!

**Pull request closed without merging**